x-pack/plugins/code/common/git_url_utils.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import GitUrlParse from 'git-url-parse';
+
+export function isValidGitUrl(
+  url: string,
+  hostWhitelist?: string[],
+  protocolWhitelist?: string[]
+): boolean {
+  const regex = /(?:git|ssh|https?|git@[-\w.]+):(\/\/)?(.*?)(\.git)?(\/?|\#[-\d\w._]+?)$/;
```

Comment:
I don't think we need this regex, or that it's really buying us anything. I think we should pass the argument to `git-url-parse` and validate the output from that library, not trying to define a regex that pre-validates it.

Reply:
done